cdap-api/src/main/java/co/cask/cdap/api/workflow/AbstractWorkflowAction.java

```diff
@@ -49,8 +50,8 @@ public void destroy() {
     // No-op
   }
 
-  protected final WorkflowContext getContext() {
-    return context;
+  public WorkflowContext getContext() {
```

Comment:
The action specs only provide the String names of the datasets.  The initialize method will create the Datasets, so I use the action to get back the instantiated Datasets